New module campaignion_expiry: Contact & submission expiry

torotil opened the pull request.

torotil committed:
campaignion_expiry: New module based on existing scripts.

torotil committed:
[expiry] Replace hard-coded values with variables.

torotil committed:
[expiry] Use little_helpers services to run the cron-jobs and pass variables.

torotil committed:
[expiry] Make keep MP data flag a variable too.

torotil committed:
[expiry] Extend logging for contact expiry.

torotil committed:
[expiry] Decrease contact batch size to 20.

katzmo commented on campaignion_expiry/campaignion_expiry.info:
```diff
@@ -0,0 +1,9 @@
+name = Mind Expiry
```
I’d suggest a more generic name, e.g. Campaignion Expiry? :wink:

katzmo commented on campaignion_expiry/campaignion_expiry.variable.inc:
```diff
@@ -0,0 +1,41 @@
+function campaignion_expiry_variable_info($options) {
+  $v['campaignion_expiry_cron_time_limit'] = [
+    'title' => t('Time limit for expiry cron-jobs'),
+    'description' => t('When an expiry cron-job has been running for more than this amount of seconds no new batc will be started during this cron-run.'),
```
batc → batch

katzmo commented on campaignion_expiry/campaignion_expiry.variable.inc:
```diff
@@ -0,0 +1,41 @@
+  ];
+  $v['campaignion_expiry_contact_time_frame'] = [
+    'title' => t('Contact expiry time frame'),
+    'description' => t('Contact’s that didn’t have any associated activity within that time-frame will be anonymized. This must be a string valid for strtotime(). Example: "-12 months" would expire submissions after a year.'),
```
"Contact’s" → "Contacts", "would expire submissions" → "would expire contacts"

katzmo commented on campaignion_expiry/src/ContactCron.php:
```diff
@@ -0,0 +1,135 @@
+  /**
+   * Create a new cron-job instance.
+   */
+  public function __construct(int $time_limit, string $inactive_since_str, bool $keep_mp_fields = FALSE) {
```
It would be nice to document the construction arguments (or use class variables, like SubmissionCron).

torotil committed:
[expiry] Better module name and description.

torotil committed:
[expiry] Add documentation for ContactCron::__construct().

torotil committed:
[expiry] Flag instead of timestammp for tracking anonymized submissions

torotil committed:
[expiry] More accurate log messages

The log message always stated that “all submissions up to …” were
expired even if the cron-job ran out of time.
